remove assert and code cleanup (#1685)
nf-core/tools: CHANGELOG.md

- Remove support for Python 3.6 ([#1680](https://github.com/nf-core/tools/pull/1680))
- Add support for Python 3.9 and 3.10 ([#1680](https://github.com/nf-core/tools/pull/1680))

> - Programm control flow no longer affected when run with optimizations ([#1685](https://github.com/nf-core/tools/pull/1685))

Reviewer: ```suggestion
- Program control flow no longer affected when run with optimizations ([#1685](https://github.com/nf-core/tools/pull/1685))
```

Author: ```suggestion
- Invoking Python with optimizations no longer affects the program control flow ([#1685](https://github.com/nf-core/tools/pull/1685))
```
